system-x/services/hashicorp-vault/src/main/java/software/tnb/hashicorp/vault/validation/HashicorpVaultValidation.java

```diff
@@ -0,0 +1,45 @@
+package software.tnb.hashicorp.vault.validation;
+
+import software.tnb.common.validation.Validation;
+
+import org.springframework.vault.authentication.TokenAuthentication;
+import org.springframework.vault.client.VaultEndpoint;
+import org.springframework.vault.core.VaultKeyValueOperations;
+import org.springframework.vault.core.VaultKeyValueOperationsSupport;
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.support.VaultResponse;
+
+import java.util.Map;
+
+public class HashicorpVaultValidation implements Validation {
+    private VaultTemplate client;
+
+    public HashicorpVaultValidation(String host, int port, String token, String scheme) {
+        VaultEndpoint vaultEndpoint = new VaultEndpoint();
+        vaultEndpoint.setHost(host);
+        vaultEndpoint.setPort(port);
+        vaultEndpoint.setScheme(scheme);
+
+        client = new VaultTemplate(
+            vaultEndpoint,
+            new TokenAuthentication(token)
+        );
```

Comment:
after changing the client in `HashicorpVault` class, creating a new client instance is usually done in `openResources` method. Then just pass the `client()` method to `HashicorpVaultValidation`, like [here](https://github.com/tnb-software/TNB/blob/main/system-x/services/jms/amq/src/main/java/software/tnb/jms/amq/service/AMQBroker.java#L36-L45) for example